tests/parsenum/main.c

```diff
@@ -16,16 +16,36 @@ main(int argc, char * argv[])
 
 	WARNP_INIT;
 
-#define TEST(x, y, z, w) do {						\
-	fprintf(stderr, "PARSENUM(\"%s\", %s, %s)\n", y, #z, #w);	\
-	if (PARSENUM(x, y, z, w))					\
-		warnp("PARSENUM");					\
-	fprintf(stderr, "%f %zu %d\n", d, s, i);			\
+#define TEST4_PROLOGUE(str, var, min, max, target)		\
+	var parsenum_x;						\
+	int parsenum_ret;					\
+	fprintf(stderr, "Parsing \"%s\" as %s between %s and %s (incl.) yields " target "... ", str, #var, #min, #max);			\
+	parsenum_ret = PARSENUM(&parsenum_x, str, min, max);
+
+#define TEST4_SUCCESS(str, var, min, max, target) do {		\
+	TEST4_PROLOGUE(str, var, min, max, #target);		\
+	if (parsenum_ret == 0) {				\
+		if ((parsenum_x >= target) && (parsenum_x <= target)) {	\
```

Comment:
ugliness due to trying to avoid floating-point comparisons.  The original code compared floats by printing them to 6 decimal places and comparing the strings.  I could still use this method, of course.  Or I could do a more standard "`fabs(x-y) < 1e-10`" thing.